contributed_definitions/nyaml/NXtransfer_matrix_table.yaml

```diff
@@ -0,0 +1,69 @@
+category: base 
+doc: |
+    Transfermatrix relationship between N input beams and M output beams.
+    It contains a table with the relevant matricis to be used for different
+    transmissitted properties (such as polarization, intensity, phase).
+
+    Data structure for all transfermatrices of an optical element in a setup.
+    For each combination of N input and M output beams and for L physical
+    concept (i.e. beam intensity), one matrix can be defined.
+
+    In this way, the transfermatrix table has the dimension NxM.
+
+    For each entry, in this transfermatrix, there are L formalisms.
+    Each formalism has the dimension math:`dim(L_i)xdim(L_i)`,
+    whereby math:`L_i` is the specific physical concept (Intensity, polarization, direction).
+
+    A beamsplitter with two input laser beams can have a total of
+    four transfermatrices (2 Input x 2 Output).
+
+    The dimension of the transfermatrix depends on the parameters.
+    Examples are:
+    1x1 for intensity/power
+    2x2 for jones formalism
+    3x3 for direction
+symbols:
+  N_variables: |
+    Dimension of the number of colums & rows for the matrix, which stores
+    a scalar (dim1) or vector (dim3)
+type: group
+NXtransfer_matrix_table(NXobject):
+  TRANSFER_MATRIX(NXdata):
+    doc: |
+      Transfermatrix relationship between N input beams and M output beams.
+      It contains a table with the relevant matricis to be used for different
+      transmissitted properties (such as polarization, intensity, phase).
+
+      Data structure for all transfermatrices of an optical element in a setup.
+      For each combination of N input and M output beams and for L physical
+      concept (i.e. beam intensity), one matrix can be defined.
+
+      In this way, the transfermatrix table has the dimension NxM.
+
+      For each entry, in this transfermatrix, there are L formalisms.
+      Each formalism has the dimension math:`dim(L_i)xdim(L_i)`,
+      whereby math:`L_i` is the specific physical concept (Intensity, polarization, direction).
+
+      A beamsplitter with two input laser beams can have a total of
+      four transfermatrices (2 Input x 2 Output).
+
+      The dimension of the transfermatrix depends on the parameters.
+      Examples are:
+      1x1 for intensity/power
+      2x2 for jones formalism
+      3x3 for direction
+    \@input:
+      doc: |
+        Specific name of input beam which the transfermatrix table is related to.
+    \@output:
+      doc: |
+        Specific name of output beam which the transfermatrix table is related to.
+    dimensions:
+      doc: |
+        Square matrix with dimension N_variables x N_variables
+      rank: 2
+      dim:
+        [
+          [1, N_variables],        
+          [2, N_variables], 
+        ]
```

Comment:
```suggestion
    data:
      dimensions:
        doc: |
          Square matrix with dimension N_variables x N_variables
        rank: 2
        dim:
          [
            [1, N_variables],        
            [2, N_variables], 
          ]
```

Reply:
Shouldn't data be of type NX_NUMBER?

Reply:
Yes, changed it

Reply:
Above is a suggestion to fix ci/cd. The problem was that there was a dimension attached to an NXdata group, which is not possible. So a field with dimension values should be introduced. However, this is a suggestion open for discussion as I don't know the details of this work.

@RonHildebrandt @sanbrock

Reply:
Yes, or rather TRANSFER_MATRIX(NX_NUMBER) if no special default plot will be attached.

Reply:
Yes, this makes more sense as this is a matrix which will typically not have any dimensional axes.